src/Systmm.f90

```diff
@@ -60,6 +61,60 @@ SUBROUTINE SYSTMM(temp_file,param_file)
 allocate (Q_na(heat_cells))
 allocate (press(heat_cells))
 allocate (wind(heat_cells))
+allocate (dt_res(2*heat_cells))
+allocate (resx(2*heat_cells))
+! 
+!   reservoir allocatable
+!
+allocate (depth_e(nres))
+allocate (depth_h(nres))
+allocate (surface_area(nres))
+allocate (T_epil(nres))
+T_epil = 10
+allocate (T_hypo(nres))
+T_hypo = 10
+allocate (stream_T_in(nres))
+stream_T_in = 10
+allocate (density_epil(nres))
+density_epil = 0
+allocate (density_hypo(nres))
+density_hypo = 0
+allocate (density_in(nres))
+density_in = 0
+allocate (volume_e_x(nres))
+allocate (volume_h_x(nres))
+allocate (dV_dt_epi(nres))
+allocate (dV_dt_hyp(nres))
+allocate (K_z(nres))
+K_z = 0.1
+allocate (temp_change_ep(nres))
+temp_change_ep = 0
+allocate (temp_change_hyp(nres))
+temp_change_hyp = 0
+allocate (res_run(nres))
+res_run = .false.
+allocate (res_start(nres))
+res_start = .false.
+allocate (T_res(nres))
+T_res = 10
+allocate (T_res_in(nres))
+T_res_in = 10
+allocate (Q_trib_tot(heat_cells))
+allocate (T_trib_tot(heat_cells))
+allocate (Q_res_in(nres))
+allocate(temp_out(nres))
+temp_out = 10
+allocate(temp_out_i(nres))
+temp_out_i = 10
+allocate (trib_res(heat_cells))
+allocate(diffusion_tot(nres))
+allocate(advec_hyp_tot(nres))
+allocate(advec_epi_tot(nres))
+allocate(qsurf_tot(nres))
+allocate(reservoir_storage(nres))
+allocate(reservoir_storage_prev(nres))
+
```

Comment:
It's better to put all those lines defining the initial values together. For example, you can put all those lines after you define all variables.